landing_page/spec/controllers/micropost_controller_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+
+RSpec.describe MicropostsController, type: :controller do
+  before :each do
+    @user = FactoryBot.create(:user)
+    session[:user_id] = @user.id
+    login(@user)
+    allow(controller).to receive(:current_user).and_return(@user)
+    @micropost = FactoryBot.create(:micropost, user_id: @user.id)
+  end
+
+  context 'create a new post with acceptable credentials' do
+    it 'successfully adds a new post' do
+      expect { create_new_micropost }.to change(Micropost, :count).by(1)
+    end
+
+    it 'redirects the user to the list of blog posts upon post creation' do
+      create_new_micropost
+      expect(response).to redirect_to(microposts_url)
+      expect(flash[:success]).to be_present
+    end
+  end
+
+  context 'GET #index' do
+    it 'lists the posts' do
+      get :index
+      expect(assigns(:microposts)).to eq([@micropost])
+      expect(response.status).to eq(200)
+    end
+  end
+
+  context 'GET #show' do
+    it 'shows the full post' do
+      get :show, params: { id: @micropost.id }
+      expect(response).to render_template('show')
+      expect(response.status).to eq(200)
+    end
+  end
+
+  context 'DELETE #destroy' do
+    it 'soft deletes a post from the database' do
+      expect { get :destroy, params: { id: @micropost.id } }.to change { Micropost.kept.count }.by(-1)
+      expect(assigns(:microposts)).to be_nil
+      expect(response).to redirect_to(microposts_url)
+      expect(flash[:warning]).to be_present
+    end
+  end
+
+  context ' GET #recover' do
+    it 'should recover a soft deleted post' do
+      get :destroy, params: { id: @micropost.id }
+      expect { get :recover, params: { id: @micropost.id } }.to change { Micropost.kept.count }.by(1)
```

Comment:
same comment as above regarding readability

Reply:
Resolved in commit 853619d7b6f01be347180ed961e9cd7e2a1dd5b8